diff -up librelp-1.10.0/src/relp.c.orig librelp-1.10.0/src/relp.c
--- librelp-1.10.0/src/relp.c.orig	2021-08-17 08:33:12.416786299 +0200
+++ librelp-1.10.0/src/relp.c	2021-08-17 08:33:45.070119507 +0200
@@ -385,9 +385,9 @@ relpEngineSetTLSLibByName(relpEngine_t *
 	}
 
 	if(!strcasecmp(name, "gnutls")) {
-		relpEngineSetTLSLib(pThis, RELP_USE_GNUTLS);
+		CHKRet(relpEngineSetTLSLib(pThis, RELP_USE_GNUTLS));
 	}else if(!strcasecmp(name, "openssl")) {
-		relpEngineSetTLSLib(pThis, RELP_USE_OPENSSL);
+		CHKRet(relpEngineSetTLSLib(pThis, RELP_USE_OPENSSL));
 	} else {
 		relpEngineCallOnGenericErr(pThis, "librelp", RELP_RET_PARAM_ERROR,
 				"invalid tls lib '%s' requested; this version of "
